examples/react/basic-default-search-params/src/main.tsx

```diff
@@ -96,31 +96,54 @@ const postsRoute = new Route({
   path: 'posts',
   loader: () => fetchPosts(),
   component: PostsComponent,
+  validateSearch: z.object({
+    sort: z.enum(['asc', 'desc']).optional(),
+  }),
 })
 
 function PostsComponent() {
   const posts = postsRoute.useLoaderData()
 
+  const [sort = 'asc', setSort] = postsRoute.useSearchState({
+    key: 'sort',
+  })
+
   return (
     <div className="p-2 flex gap-2">
       <div className="list-disc bg-gray-800/70 rounded-lg divide-y divide-green-500/30">
-        {posts.map((post, index) => {
-          return (
-            <div key={post.id} className="whitespace-nowrap">
-              <Link
-                to={postRoute.to}
-                search={{
-                  postId: post.id,
-                  color: index % 2 ? 'red' : undefined,
-                }}
-                className="block py-1 px-2 text-green-300 hover:text-green-200"
-                activeProps={{ className: '!text-white font-bold' }}
-              >
-                <div>{post.title.substring(0, 20)}</div>
-              </Link>
-            </div>
+        <select
+        className='bg-gray-800/70 rounded-lg w-full'
+          value={sort}
+          onChange={(e) => setSort(e.target.value as any)}
+        >
+          <option value="asc">Asc</option>
+          <option value="desc">Desc</option>
+        </select>
+
+        {/* TODO: probably better to drive the loader with search but need to research how */}
```

Comment:
This is probably worth doing as it would be a little more correct to how the example is written. Need to check the docs!